### PR TITLE
backend: add ability to deprecate enumeration values

### PR DESCRIPTION
**Pull request by kuceram**

### CODE REVIEW

**Comment by kuceram** [comment]
This is not ready yet, I've just created this pull request in order to test my changes on travis... 🎢

**Comment by musketyr** [comment]
@kuceram ok, but the test runs for every branch even if you don't create new pull request

**Comment by kuceram** [comment]
@musketyr roger...

File: ModelCatalogueCorePlugin/grails-app/controllers/org/modelcatalogue/core/EnumeratedTypeController.groovy [hunk 1]
```diff
@@ -39,6 +41,24 @@ class EnumeratedTypeController extends DataTypeController<EnumeratedType> {
...
+        if (!enumeratedType) {
+            notFound()
+            return
+        }
+
+        def enumerations = enumeratedType.enumerationsObject
+        println enumerations.toJsonString()
+        enumeratedType.setEnumerations(enumerations.withDeprecatedEnumeration(jsonPayload.enumerationId, jsonPayload.deprecated))
+
+        enumeratedType.save(failOnError: true)
```
Comment (musketyr) [inline ModelCatalogueCorePlugin/grails-app/controllers/org/modelcatalogue/core/EnumeratedTypeController.groovy]: better is to do 

```
if (enumeratedType.save(flush: true))  {
    return respond(enumeratedType, [status: OK])
}
respond enumeratedType.errors
```

Comment (kuceram) [inline ModelCatalogueCorePlugin/grails-app/controllers/org/modelcatalogue/core/EnumeratedTypeController.groovy]: can do...

**Pull request merged**